cve_bin_tool/checkers/libexif.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+import re
+
+"""
+CVE checker for libexif
+
+References:
+https://libexif.github.io/
+https://www.cvedetails.com/vulnerability-list/vendor_id-2861/Libexif.html
+
+
+Note: Some of the "first vulnerable in" data may not be entered correctly.
+"""
+from ..util import regex_find
+
+
+def guess_contains_libexif(lines):
+    """Tries to determine if a file includes libexif.
+    """
+    for line in lines:
+        if "An unknown option was passed in to libcurl" in line:
+            return 1
+        if (
+            "Most digital cameras produce EXIF files, which are JPEG files with extra tags that contain information about the image. The EXIF library allows you to parse an EXIF file and read the data from those tags."
+            in line
+        ):
+            return 1
+        if "libexif-12.mo" in line:
+            return 1
+    return 0
+
+
+def findOccurance(lines, *args):
+    patterns = [re.compile(regex) for regex in args]
+    for line in lines:
+        print(line)
+        for pattern in patterns:
+            match = pattern.search(line)
+            if match:
+                return match.group(0)
+    return None
+
+
+def get_version(lines, filename):
+    """returns version information for libexif as found in a given file.
+    The version info is returned as a tuple:
+        [modulename, is_or_contains, version]
+
+    VPkg: libexif_project, libexif
+    """
+    regex = [r"(\d+\.)(\d+\.)(\d)(\d)"]
```

Comment:
This is going to hit a lot of other numbers that may not be a libexif version number.  Is there anything we can do to narrow it down to something specific to exif?  Without some way to cut down on false positives, I don't think this checker is going to be usable.  (You'll see that this has happened with some other things we tried to detect too if you look at the other things tagged as "signature needs improvement")